godot-core/src/init/mod.rs

```diff
@@ -144,7 +144,9 @@ unsafe fn gdext_on_level_init(level: InitLevel) {
         }
         InitLevel::Editor => {
             #[cfg(all(since_api = "4.3", feature = "docs"))]
-            crate::docs::register();
+            unsafe {
+                crate::docs::register();
+            }
```

Comment:
Why this change?

Reply:
I can't compile godot-core without declaring this function call as unsafe :shrug:. 

```cmd
irwin@toster:~/(…)/missing_docs/bug-repro-godot-rust-docs-master/docstest$ cargo build
   Compiling godot-macros v0.1.3 (/home/irwin/apps/godot/opensource-contr/missing_docs/gdext/godot-macros)
   Compiling godot-core v0.1.3 (/home/irwin/apps/godot/opensource-contr/missing_docs/gdext/godot-core)
error: call to unsafe function `docs::register` is unsafe and requires unsafe block (error E0133)
   --> /home/irwin/apps/godot/opensource-contr/missing_docs/gdext/godot-core/src/init/mod.rs:147:13
    |
147 |             crate::docs::register();
    |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
    |
    = note: for more information, see issue #71668 <https://github.com/rust-lang/rust/issues/71668>
    = note: consult the function's documentation for information on how to avoid undefined behavior
note: an unsafe function restricts its caller, but its body is safe by default
   --> /home/irwin/apps/godot/opensource-contr/missing_docs/gdext/godot-core/src/init/mod.rs:132:1
    |
132 | unsafe fn gdext_on_level_init(level: InitLevel) {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: the lint level is defined here
   --> /home/irwin/apps/godot/opensource-contr/missing_docs/gdext/godot-core/src/init/mod.rs:131:8
    |
131 | #[deny(unsafe_op_in_unsafe_fn)]
    |        ^^^^^^^^^^^^^^^^^^^^^^

error: could not compile `godot-core` (lib) due to 1 previous error

```

Reply:
call to `crate::docs::register()` is unsafe if experimental-threads feature is enabled. 

Should I create two variants of `crate::docs::register()` that depends on set features?

Reply:
Call to `crate::docs::register();` is unsafe and I can't compile test project without explicitly marking this call as unsafe :thinking: .

```rust
irwin@toster:~/apps/godot/opensource-contr/missing_docs/bug-repro-godot-rust-docs-master/docstest$ cargo build
   Compiling godot-core v0.1.3 (/home/irwin/apps/godot/opensource-contr/missing_docs/gdext/godot-core)
error: call to unsafe function `docs::register` is unsafe and requires unsafe block (error E0133)
   --> /home/irwin/apps/godot/opensource-contr/missing_docs/gdext/godot-core/src/init/mod.rs:147:17
    |
147 |                 crate::docs::register();
    |                 ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
    |
    = note: for more information, see issue #71668 <https://github.com/rust-lang/rust/issues/71668>
    = note: consult the function's documentation for information on how to avoid undefined behavior
note: an unsafe function restricts its caller, but its body is safe by default
   --> /home/irwin/apps/godot/opensource-contr/missing_docs/gdext/godot-core/src/init/mod.rs:132:1
    |
132 | unsafe fn gdext_on_level_init(level: InitLevel) {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: the lint level is defined here
   --> /home/irwin/apps/godot/opensource-contr/missing_docs/gdext/godot-core/src/init/mod.rs:131:8
    |
131 | #[deny(unsafe_op_in_unsafe_fn)]
    |        ^^^^^^^^^^^^^^^^^^^^^^

error: could not compile `godot-core` (lib) due to 1 previous error
```

```rust
/missing_docs/gdext/godot$ cargo test --features register-docs,api-custom,godot-core/docs,godot-core/api-custom -- --nocapture
   Compiling godot-macros v0.1.3 (/home/irwin/apps/godot/opensource-contr/missing_docs/gdext/godot-macros)
   Compiling godot-core v0.1.3 (/home/irwin/apps/godot/opensource-contr/missing_docs/gdext/godot-core)
error: call to unsafe function `docs::register` is unsafe and requires unsafe block (error E0133)
   --> godot-core/src/init/mod.rs:147:13
    |
147 |             crate::docs::register();
    |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
    |
    = note: for more information, see issue #71668 <https://github.com/rust-lang/rust/issues/71668>
    = note: consult the function's documentation for information on how to avoid undefined behavior
note: an unsafe function restricts its caller, but its body is safe by default
   --> godot-core/src/init/mod.rs:132:1
    |
132 | unsafe fn gdext_on_level_init(level: InitLevel) {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: the lint level is defined here
   --> godot-core/src/init/mod.rs:131:8
    |
131 | #[deny(unsafe_op_in_unsafe_fn)]
    |        ^^^^^^^^^^^^^^^^^^^^^^

error: could not compile `godot-core` (lib) due to 1 previous error
```

Reply:
Are you using rust 2024?

Reply:
Somebody must have put that deny in there and forgotten to check the #[cfg]d out things i suppose...

Reply:
@bend-n it seems so 
![image](https://github.com/user-attachments/assets/96bbebab-86c8-4ee2-86f3-6a3add27c37c)

Reply:
Good remark, I'll make sure this feature is enabled in at least one CI run.

For now, you can add the `unsafe` block and prepend it with a safety assertion line:
```rs
// SAFETY: binding has been initialized at this point.
```

Reply:
Done: https://github.com/godot-rust/gdext/pull/819
I can later rebase this PR onto `master`, to make sure the CI passes.